src/PKSim.Core/Mappers/ParameterListOfValuesRetriever.cs

```diff
@@ -0,0 +1,88 @@
+         {
+            listOfValues.Add(CoreConstants.Parameters.PARTICLE_SIZE_DISTRIBUTION_NORMAL, PKSimConstants.UI.Normal);
+            listOfValues.Add(CoreConstants.Parameters.PARTICLE_SIZE_DISTRIBUTION_LOG_NORMAL, PKSimConstants.UI.LogNormal);
+         }
+         else if (parameter.IsNamed(CoreConstants.Parameters.PLASMA_PROTEIN_BINDING_PARTNER))
+         {
+            listOfValues.Add(CoreConstants.Compound.BINDING_PARTNER_ALBUMIN, PKSimConstants.UI.Albumin);
+            listOfValues.Add(CoreConstants.Compound.BINDING_PARTNER_AGP, PKSimConstants.UI.Glycoprotein);
+            listOfValues.Add(CoreConstants.Compound.BINDING_PARTNER_UNKNOWN, PKSimConstants.UI.Unknown);
+         }
+         else if (parameter.IsNamed(CoreConstants.Parameters.NUMBER_OF_BINS))
+         {
+            addNumericListOfValues(listOfValues, 1, CoreConstants.Parameters.MAX_NUMBER_OF_BINS);
+         }
+         else if (parameter.NameIsOneOf(CoreConstants.Parameters.Halogens))
+         {
+            addNumericListOfValues(listOfValues, 0, CoreConstants.Parameters.MAX_NUMBER_OF_HALOGENS);
+         }
+         else if (parameter.Name.StartsWith(CoreConstants.Parameters.ParameterCompoundTypeBase))
+         {
+            listOfValues.Add(CoreConstants.Compound.COMPOUND_TYPE_ACID, CompoundType.Acid.ToString());
+            listOfValues.Add(CoreConstants.Compound.COMPOUND_TYPE_NEUTRAL, CompoundType.Neutral.ToString());
+            listOfValues.Add(CoreConstants.Compound.COMPOUND_TYPE_BASE, CompoundType.Base.ToString());
+         }
+         else if (parameter.IsNamed(CoreConstants.Parameters.PARTICLE_DISPERSE_SYSTEM))
+         {
+            listOfValues.Add(CoreConstants.Parameters.MONODISPERSE, PKSimConstants.UI.Monodisperse);
+            listOfValues.Add(CoreConstants.Parameters.POLYDISPERSE, PKSimConstants.UI.Polydisperse);
+         }
+         else if (parameter.IsNamed(CoreConstants.Parameters.PRECIPITATED_DRUG_SOLUBLE))
+         {
+            listOfValues.Add(CoreConstants.Parameters.SOLUBLE, PKSimConstants.UI.Soluble);
+            listOfValues.Add(CoreConstants.Parameters.INSOLUBLE, PKSimConstants.UI.Insoluble);
+         }
+         else if (parameter.IsNamed(CoreConstants.Parameters.GESTATIONAL_AGE))
+         {
+            addNumericListOfValues(listOfValues, CoreConstants.PretermRange.Min(), CoreConstants.PretermRange.Max());
+         }
+         else if (CoreConstants.Parameters.AllBooleanParameters.Contains(parameter.Name))
+         {
+            listOfValues.Add(1, PKSimConstants.UI.Yes);
+            listOfValues.Add(0, PKSimConstants.UI.No);
+         }
+         else if (parameter.NameIsOneOf(CoreConstants.Parameters.PARA_ABSORBTION_SINK, CoreConstants.Parameters.TRANS_ABSORBTION_SINK))
+         {
+            listOfValues.Add(CoreConstants.Parameters.SINK_CONDITION, PKSimConstants.UI.SinkCondition);
+            listOfValues.Add(CoreConstants.Parameters.NO_SINK_CONDITION, PKSimConstants.UI.NoSinkCondition);
+         }
```

Comment:
Missing exception at the end
(Parameter has list of values but nothing was added to the listOfValues-Cache)

Reply:
Yeah. Don't know why would throw an exception to be honest. I think in that case the parameter behaves as a normal parameter like in MoBi.

Reply:
But I am ok to put it back. No strong opinion here for once. This would be more of a dev error really (which We never do;-))

Reply:
exactly :)

Reply:
ok updated. And even wrote tests :